src/neptune_scale/sync/metadata_splitter.py

```diff
@@ -219,3 +215,14 @@ def _skip_non_finite(self, step: Optional[float | int], metrics: dict[str, float
                     raise NeptuneFloatValueNanInfUnsupported(metric=k, step=step, value=v)
 
             yield k, v
+
+    def _make_empty_update_snapshot(self) -> UpdateRunSnapshot:
+        include_preview = self._metrics_data and self._metrics_preview
+        return UpdateRunSnapshot(
+            step=self._step,
```

Comment:
I'm tempted to also write step iff there are metrics present (similar to how I do preview). In practice it doesn't matter much as step will be None, unless some metrics are provided; only case I can think of where it'd make a difference is when a huge write of metrics and configs/tags gets split over multiple messages.

Reply:
Unless I'm missing something here, we're talking about 8 bytes in a 1MB (and we'll increase that limit) msg. So negligible, right?

Reply:
Correct. This is why I haven't actually done it - the benefit is negligible and it is introducing some delta in behavior (so - some level of risk).